src/daft-recordbatch/src/ops/window_states/mod.rs

```diff
@@ -0,0 +1,106 @@
+mod count;
+mod count_distinct;
+mod max;
+mod mean;
+mod min;
+mod sum;
+
+use std::cmp::{Eq, Ordering};
+
+use common_error::{DaftError, DaftResult};
+use count::CountWindowState;
+use count_distinct::CountDistinctWindowState;
+use daft_core::prelude::*;
+use daft_dsl::AggExpr;
+use max::MaxWindowState;
+use min::MinWindowState;
+use sum::SumWindowState;
+
+/// Trait for window aggregation state implementations
+pub trait WindowAggStateOps {
+    /// Add a value to the state with index information
+    fn add(&mut self, start_idx: usize, end_idx: usize) -> DaftResult<()>;
+
+    /// Remove a value from the state with index information
+    fn remove(&mut self, start_idx: usize, end_idx: usize) -> DaftResult<()>;
+
+    /// Evaluate the current state and push the result to internal buffer
+    fn evaluate(&mut self) -> DaftResult<()>;
+
+    /// Build the final result series containing all accumulated results
+    fn build(&self) -> DaftResult<Series>;
+}
+
+#[derive(Debug, Clone)]
+pub struct IndexedValue {
+    pub value: Series,
+    pub idx: u64,
+}
```

Comment:
Instead of storing both an index and a value in IndexedValue, could we just store the index and a reference to the entire series? Then, slice the series during the comparisons. That should reduce the amount of data that needs to be stored at a time.